transport/grpc/server.go

```diff
@@ -150,21 +156,6 @@ func NewServer(opts ...ServerOption) *Server {
 // examples:
 //   grpc://127.0.0.1:9000?isSecure=false
 func (s *Server) Endpoint() (*url.URL, error) {
-	s.once.Do(func() {
-		lis, err := net.Listen(s.network, s.address)
```

Comment:
would be it satisfy if just replace this line by s.lis(if s.lis is not nil)?

Reply:
Yes，I want to get rid of the side effects of endpoint。